contracts/WithdrawalsReserves.sol

```diff
@@ -0,0 +1,79 @@
+pragma solidity ^0.4.8;
+
+import "./Owned.sol";
+import "./ServicesI.sol";
+import "./vendor/MathLib.sol";
+import "./vendor/SafeSendLib.sol";
+
+/* Central reserves for all value in flight
+ * includes: requested withdrawals (emitter excess), contract values after
+ *           dissolution, participant funds before match
+ */
+contract WithdrawalsReserves is Owned {
+    using MathLib for uint;
+    using SafeSendLib for address;
+
+    mapping (address => uint) public balances;
+
+    address servicesAddress;
+
+    event Reserve(address participant, uint amount);
+    event Withdraw(address participant, uint amount);
+    event Transfer(address participant, uint amount);
+
+    function WithdrawalsReserves(address _servicesAddress) {
+        servicesAddress = _servicesAddress;
+    }
+
+    modifier fromFactory() {
+        ServicesI services = ServicesI(servicesAddress);
+        address factoryAddress = services.serviceAddress(sha3("Factory"));
+        if (msg.sender != factoryAddress) { throw; }
+        _;
+    }
+
+    function reserve(address _participant) payable {
+        uint amount = msg.value;
+        balances[_participant] = balances[_participant].safeAdd(amount);
+
+        Reserve(_participant, amount);
+    }
+
+    function withdraw() {
+        address participant = msg.sender;
+        uint amount = balances[participant];
+
+        // re-entry protection
+        balances[participant] = 0;
+
+        uint sent = participant.safeSend(amount);
+
+        uint newBalance = amount.flooredSub(sent);
+        balances[participant] = newBalance;
+
+        Withdraw(participant, sent);
+    }
+
+    function transfer(
+        uint _amount,
+        address _from,
+        address _contractAddress,
+        bytes32 _funcSigHash)
+        fromFactory
+        returns (uint sent)
+    {
+        uint originalBalance = balances[_from];
+
+        if (originalBalance < _amount) throw;
+
+        // re-entry protection
+        balances[_from] = 0;
+
+        if (!_contractAddress.call.value(_amount)(bytes4(_funcSigHash))) throw;
+
+        uint newBalance = originalBalance.flooredSub(_amount);
+        balances[_from] = newBalance;
+
+        Transfer(_from, _amount);
+    }
```

Comment:
quoth @chatch:
> Does the BVC checkMargin need to get called on this initial deposit? Maybe deposit function could be split into 2 and have an initalDeposit that just accepts ether transfer. Then you could constrain the gas on the generic call to BVC deposit as well for a little more safety. An initalDeposit function could further enforce msg.sender is WithdrawalReserves to block any other possible call to it.
> 
> Maybe intentional because margins calculations aren't there yet but the first line in checkMargin is just "_;" so the rest of the logic gets ignored.